### PR TITLE
dotnet-watch tutorial

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/Docs/issues/696

A tutorial on using `dotnet watch`

Please review @Rick-Anderson @rachelappel @danroth27

### CODE REVIEW

File: aspnet/tutorials/dotnet-watch.rst [hunk 1]
```diff
@@ -0,0 +1,198 @@
+.. _dotnet-watch:
+
+Developing ASP.NET applications using dotnet watch
```
Comment (BrennanConroy) [inline aspnet/tutorials/dotnet-watch.rst]: ASP.NET Core

File: aspnet/tutorials/dotnet-watch.rst [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+1.  ``dotnet restore``
+2.  ``cd WebApp``
+3.  ``dotnet run``
+
+The console output will show messages similar to the ones below, indicating that the application is now running and waiting for requests:
+
+.. code-block:: bash
+
+  user$ dotnet run
```
Comment (cesarbs) [inline aspnet/tutorials/dotnet-watch.rst]: Remove `user` and leave it just as `$`.

File: aspnet/tutorials/dotnet-watch.rst [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+  log  : Restoring packages for /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/project.json...
+  log  : Restoring packages for tool 'Microsoft.DotNet.Watcher.Tools' in /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp/project.json...
+  log  : Installing Microsoft.DotNet.Watcher.Core 1.0.0-preview2-final.
+  log  : Installing Microsoft.DotNet.Watcher.Tools 1.0.0-preview2-final.
+
+Running ``dotnet`` commands using ``dotnet watch``
+--------------------------------------------------
+
+Any ``dotnet`` command can be run through ``dotnet watch`` by simply adding ``watch`` after ``dotnet``. For example:
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: Any ``dotnet`` command can be run through ``dotnet watch`` by  appending  ```: dotnet watch <cmd>``

File: aspnet/tutorials/dotnet-watch.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+Making changes with ``dotnet watch``
+------------------------------------
+
+Let's fix the bug that we discovered when we tried to compute the product of two number.
+
+Open ``WebApp/Controllers/MathController.cs`` in a text editor.
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: ~~in a text editor~~

File: aspnet/tutorials/dotnet-watch.rst [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+You'll notice that we've intentionally introduced a bug in the code.
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: ~~You'll notice that ~~

File: aspnet/tutorials/dotnet-watch.rst [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  Time elapsed 00:00:03.3312829
+  
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+
+
+Refresh or open the webpage pointing to ``http://localhost:5000/api/math/product?a=4&b=5`` and you should get the correct result (20).
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: Verify ``http://localhost:5000/api/math/product?a=4&b=5`` returns the correct result.
[don't need to tell them it's 20]

File: aspnet/tutorials/dotnet-watch.rst [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+Running tests using ``dotnet watch``
+------------------------------------
+
+So far we only used ``dotnet watch`` to run our application. The file watcher can also be used to run other ``dotnet`` commands like ``test`` or ``publish``.
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: ~~So far we only used ``dotnet watch`` to run our application.~~

Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: The file watcher can ~~also be used to~~ run other

File: aspnet/tutorials/dotnet-watch.rst [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+    Discovered:  WebAppTests
+    Starting:    WebAppTests
+    Finished:    WebAppTests
+  === TEST EXECUTION SUMMARY ===
+     WebAppTests  Total: 2, Errors: 0, Failed: 0, Skipped: 0, Time: 0.259s
+  SUMMARY: Total: 1 targets, Passed: 1, Failed: 0.
+  [DotNetWatcher] info: dotnet exit code: 0
+  [DotNetWatcher] info: Waiting for a file to change before restarting dotnet...
+
+In either case, once all the tests run, the watcher will indicate that it waits for a file to change before restarting ``dotnet test``.
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: ~~In either case,~~

Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: How's this?
Once all the tests run, the watcher will indicate that it's waiting for a file to change before restarting ``dotnet test``.

File: aspnet/tutorials/dotnet-watch.rst [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+
+3. Open the controller file in ``WebApp/Controllers/MathController.cs`` and change some code. If you haven't fixed the product bug, do it now. Save the file.
```
Comment (Rick-Anderson) [inline aspnet/tutorials/dotnet-watch.rst]: files are italic, not code
Open the controller file in *WebApp/Controllers/MathController.cs*
I probably missed other files.

**Comment by Rick-Anderson** [comment]
@victorhurdugaci  really excellent!  Let me know when you're finished or if you need anything from me.

File: aspnet/tutorials/dotnet-watch.rst [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+  [DotNetWatcher] info: Running dotnet with the following arguments: run
+  [DotNetWatcher] info: dotnet process id: 39746
+  Project WebApp (.NETCoreApp,Version=v1.0) was previously compiled. Skipping compilation.
+  Hosting environment: Production
+  Content root path: /Users/user/dev/aspnet/Docs/aspnet/tutorials/dotnet-watch/sample/WebApp
+  Now listening on: http://localhost:5000
+  Application started. Press Ctrl+C to shut down.
+
+Making changes with ``dotnet watch``
+------------------------------------
```
Comment (BrennanConroy) [inline aspnet/tutorials/dotnet-watch.rst]: Here you are starting another section, you should say something about "make sure you have dotnet watch running on the WebApp"

**Comment by victorhurdugaci** [comment]
Updated.

@Rick-Anderson any more comments? Also, should I be merged in master? Do I merge it or will you do it?

**Comment by Rick-Anderson** [comment]
looks good. :shipit:

**Comment by Rick-Anderson** [comment]
@rachelappel can you get this migrated to master? cc @victorhurdugaci  this is not showing up in master.

**Comment by rachelappel** [comment]
Merged 7597448cd5f3febe4934fceee515753a0c1fe258

**Pull request merged**